### PR TITLE
[WIP] ember-decorators

### PR DESCRIPTION
**Pull request by dwickern**
I'm creating this as a starting point for `ember-decorators` typings. I'm putting the typings here to leverage DT's test framework. Ultimately these typings belong in the ember-decorators repo.

I've found that typescript decorators are severely limited in what they can express. I'll try to document some of those issues and possible workarounds.

I based this branch on the `new-object-model` branch since the typings didn't work on master (without `noImplicitAny` or some extra type annotations).

### CODE REVIEW

**Comment by bartocc** [comment]
Hey @dwickern 

I'm willing to help here ! Do you need me to ping you on slack ?

**Comment by dwickern** [comment]
@bartocc I'd appreciate the help! I made this branch to get things started for anyone who wants to contribute. We're in slack if you have any questions

File: types/ember-decorators/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.4
+
+declare module 'ember-decorators/object/computed' {
+    export function alias(dependentKey: string): PropertyDecorator;
+    export function and(...keys: string[]): PropertyDecorator;
+}
+
+declare module 'ember-decorators/object' {
+    export function computed(...keys: string[]): MethodDecorator;
```
Comment (mike-north) [inline types/ember-decorators/index.d.ts]: needs to be a `PropertyDecorator` as well

**Comment by championswimmer** [comment]
@dwickern 
Can I help here in any way ?
Would love to see this merged

**Comment by dwickern** [comment]
I'm not sure this issue is relevant any more given that ember decorators are written in TS now. @pzuraq ?

**Comment by pzuraq** [comment]
ember-decorators are not actually written in TS, they are compatible now. If we can add the typings there I’m completely ok with it, either way

**Comment by chriskrycho** [comment]
Yeah, we should close this issue. You can check the pull requests on the specific repositories I opened over the weekend!

**Pull request closed without merging**